File: result/metagpt/claude/humaneval/137.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

typedef enum {
    TYPE_INT,
    TYPE_DOUBLE,
    TYPE_STRING
} ValueType;

typedef struct {
    ValueType type;
    union {
        int int_value;
        double double_value;
        char* string_value;
    };
} AnyValue;

AnyValue compare_one(AnyValue a, AnyValue b) {
    double numa, numb;
    AnyValue out;

    // Convert a to double
    if (a.type == TYPE_STRING) {
        char* s = a.string_value;
        char* comma = strchr(s, ',');
        if (comma) {
            *comma = '.';
        }
        numa = atof(s);
    } else if (a.type == TYPE_INT) {
        numa = (double)a.int_value;
    } else if (a.type == TYPE_DOUBLE) {
        numa = a.double_value;
    }

    // Convert b to double
    if (b.type == TYPE_STRING) {
        char* s = b.string_value;
        char* comma = strchr(s, ',');
        if (comma) {
            *comma = '.';
        }
        numb = atof(s);
    } else if (b.type == TYPE_INT) {
        numb = (double)b.int_value;
    } else if (b.type == TYPE_DOUBLE) {
        numb = b.double_value;
    }

    // Compare and return the larger value
    if (numa == numb) {
        out.type = TYPE_STRING;
        out.string_value = strdup("None");
        return out;
    } else if (numa < numb) {
        return b;
    } else {
        return a;
    }
}

File: result/metagpt/gpt4o/humaneval/137.rs
#[derive(Debug, Clone)]
enum ValueType {
    Int,
    Double,
    String,
}

#[derive(Debug, Clone)]
struct AnyValue {
    value_type: ValueType,
    int_value: Option<i32>,
    double_value: Option<f64>,
    string_value: Option<String>,
}

fn compare_one(a: AnyValue, b: AnyValue) -> AnyValue {
    let numa = match a.value_type {
        ValueType::String => {
            let mut s = a.string_value.unwrap();
            if let Some(comma_pos) = s.find(',') {
                s.replace_range(comma_pos..=comma_pos, ".");
            }
            s.parse::<f64>().unwrap_or(0.0)
        }
        ValueType::Int => a.int_value.unwrap() as f64,
        ValueType::Double => a.double_value.unwrap(),
    };

    let numb = match b.value_type {
        ValueType::String => {
            let mut s = b.string_value.unwrap();
            if let Some(comma_pos) = s.find(',') {
                s.replace_range(comma_pos..=comma_pos, ".");
            }
            s.parse::<f64>().unwrap_or(0.0)
        }
        ValueType::Int => b.int_value.unwrap() as f64,
        ValueType::Double => b.double_value.unwrap(),
    };

    if numa == numb {
        AnyValue {
            value_type: ValueType::String,
            int_value: None,
            double_value: None,
            string_value: Some("None".to_string()),
        }
    } else if numa < numb {
        b
    } else {
        a
    }
}

fn main() {
    let a = AnyValue {
        value_type: ValueType::String,
        int_value: None,
        double_value: None,
        string_value: Some("123,45".to_string()),
    };

    let b = AnyValue {
        value_type: ValueType::Double,
        int_value: None,
        double_value: Some(123.46),
        string_value: None,
    };

    let result = compare_one(a.clone(), b.clone());
    println!("Larger value: {:?}", result);
}